tests/testthat/test-module_nested_tabs.R

```diff
@@ -223,7 +285,7 @@ testthat::test_that("srv_nested_tabs.teal_module doesn't pass filter_panel_api i
     moduleServer(id, function(input, output, session) checkmate::assert_class(filter_panel_api, "FilterPanelAPI"))
   })
 
-  testthat::expect_error(
+  testthat::expect_warning(
```

Comment:
I could not find a better way to improve this test. It passes but throws an error internally. 
The test does not really test what it claims it is testing.

Reply:
So I wonder if using `rlang::env_has(environment(), "data", inherit = FALSE)` (or datasets, or filterpanelapi) would help here? - I'll test

Reply:
for the not passed case we want this:
```
moduleServer(id, function(input, output, session) {
      checkmate::assert_false(
        tryCatch(
          checkmate::test_class(data, "tdata"),
          error = function(cond) FALSE
        )
      )
    })
 ```
and you replace data with datasets/filter_panel_api

for the it is passed case we can have assert_class() directly
and then expect_no_error (or expect_error(..., NA)) for the testserver

Reply:
done.